programs/lz4cli.c

```diff
@@ -658,7 +658,11 @@ int main(int argc, const char** argv)
         if (!output_filename) output_filename = stdoutmark;
     }
     else{
+#ifdef UTIL_HAS_CREATEFILELIST
```

Comment:
I'm surprised about this need.
`UTIL_HAS_CREATEFILELIST` is used above to ensure that `recursive` can never be set to `1`. So this particular test should not need to be gated ?

Reply:
`recursive` is not declared when `UTIL_HAS_CREATEFILELIST` isn't used: see [here](https://github.com/lz4/lz4/blob/dev/programs/lz4cli.c#L330)

Reply:
oh , right !
It's designed differently from `zstd`, where `recursive` is always declared, and is just set to `0`, avoiding further preprocessor selection.

Anyway, that's probably a topic for another change.